modules/nf-core/universc/main.nf

```diff
@@ -33,26 +27,20 @@ process UNIVERSC {
     }
     def args = task.ext.args ?: ''
     def sample_arg = meta.samples.unique().join(",")
-    def reference_name = reference.name
-    def input_reads = meta.single_end ? "--file $reads" : "-R1 ${reads[0]} -R2 ${reads[1]}"
+    if ( reads instanceof Path || reads.size() != 2 ) {
+        error "UNIVERSC module only supports 2 paired end read files. Input: ${reads}"
+    }
     """
     universc \\
-        --id 'sample-${meta.id}' \\
-        ${input_reads} \\
-        --technology '${meta.technology}' \\
-        --chemistry '${meta.chemistry}' \\
-        --reference ${reference_name} \\
-        --description ${sample_arg} \\
+        --id ${meta.id} \\
+        --read1 ${reads[0]} --read2 ${reads[1]} \\
+        --reference ${reference} \\
         --jobmode "local" \\
         --localcores ${task.cpus} \\
         --localmem ${task.memory.toGiga()} \\
-        --per-cell-data \\
-        $args 1> _log 2> _err
+        $args
 
-    # save log files
     echo !! > sample-${meta.id}/outs/_invocation
```

Comment:
What bash magic is this?

Reply:
`!!` stands for the last executed command

Reply:
do we need it? If someone wants to debug they can look into `.command.sh`